lib/users.ts

```diff
@@ -104,3 +105,24 @@ export const adminRole = async (isAdmin: boolean, userId: string): Promise<[bool
     return [false, false];
   }
 };
+
+/**
+ * Retrieves the accessLog entry for the last login for a user
+ * @param userId
+ * @returns AccessLog object
+ */
+export const userLastLogin = async (userId: string): Promise<AccessLog | null> => {
+  try {
+    return await prisma.accessLog.findFirst({
+      where: {
+        userId: userId,
+        action: LoggingAction.LOGIN,
+      },
+      orderBy: {
+        timestamp: "desc",
+      },
+    });
+  } catch {
+    return null;
```

Comment:
Is there a reason not to use `prismaErrors(e,null)` ? if an error occurs it might be helpful for debugging this feature.

Reply:
@falila is correct. That function should be use since it will `return null` in the end but it will also take care of logging any Prisma errors

Reply:
Nice catch. I wasn't aware of that custom function.